test/test_unary_ufuncs.py

```diff
@@ -93,26 +120,43 @@ def generate_numeric_tensors(device, dtype, *,
                    make_tensor(large_size, device=device, dtype=dtype, low=None, high=None))
         return tensors
 
+    low = domain[0] if domain[0] is not None else -float('inf')
+    high = domain[1] if domain[1] is not None else float('inf')
+
     # Acquires dtype-specific vals
     if dtype.is_floating_point or dtype.is_complex:
-        large_vals = _large_float_vals if include_large_values else tuple()
-        extremals = _float_extremals if include_extremal_values else tuple()
-        vals = _float_vals + large_vals + extremals
-
-        # Converts float -> complex vals if dtype is complex
+        # NOTE: large and extremal complex values are not tested
+        large_vals = _large_float_vals if include_large_values and dtype.is_floating_point else tuple()
+        opt_inf_vals = (-float('inf'), float('inf')) if dtype.is_floating_point else tuple()
+        opt_nan = (float('nan'),) if dtype.is_floating_point else tuple()
+
+        # NOTE: float (and complex) filter is strictly less than / greater than
+        #   since float domains exclude their endpoints
+        unfiltered_vals = _float_vals + large_vals + opt_inf_vals
+        filtered_vals = tuple(val for val in unfiltered_vals if (val > low and val < high))
+        vals = filtered_vals + opt_nan
+
+        # Derives complex values from float values if the dtype is complex
+        # NOTE: complex values are constructed as float val x float val pairs
         if dtype.is_complex:
             vals = tuple(complex(x, y) for x, y in product(vals, vals))
     elif dtype is torch.uint8:
         vals = _unsigned_int_vals
     else:  # dtypes is a signed integer type
         assert dtype in (torch.int8, torch.int16, torch.int32, torch.int64)
-        large_vals = _large_int_vals if include_large_values else tuple()
-        vals = _int_vals + large_vals
+        unfiltered_vals = _int_vals + _large_int_vals
+
+        # filters to symmetric integer range to avoid ASAN issues
+        # NOTE: for example, abs(-128.int8) will throw a representable value ASAN error in
```

Comment:
comment looks unfinished, what does "in" refer to?

Reply:
You're right. "in" should just be removed.